tests/io/hdf5_test.py

```diff
@@ -216,6 +218,48 @@ def test_dataset():
     check_roundtrip(d)
 
 
+def test_dataset_item_can_be_read_as_data_array():
+    ds = sc.Dataset(data={'a': array_1d, 'b': array_2d})
+    with tempfile.TemporaryDirectory() as path:
+        name = f'{path}/test.hdf5'
+        ds.to_hdf5(filename=name)
+        loaded = sc.Dataset()
+        with h5py.File(name, 'r') as f:
+            for entry in f['entries'].values():
+                da = sc.io.hdf5.HDF5IO.read(entry)
+                loaded[da.name] = da
+        assert sc.identical(loaded, ds)
+
+
+def test_dataset_with_many_coords():
+    rows = 10000
+    a = sc.ones(dims=['row'], shape=[rows], dtype='float64')
+    b = sc.ones(dims=['row'], shape=[rows], dtype='float64')
+    coords = {
+        k: sc.ones(dims=['row'], shape=[rows], dtype='float64')
+        for k in 'abcdefghijklmnopqrstuvwxyz'
+    }
+    ds1 = sc.Dataset(coords=coords)
+    ds1['a'] = a
+    ds2 = sc.Dataset(coords=coords)
+    ds2['a'] = a
+    ds2['b'] = b
+    with tempfile.TemporaryDirectory() as path:
+        name1 = f'{path}/test1.hdf5'
+        name2 = f'{path}/test2.hdf5'
+        ds1.to_hdf5(filename=name1)
+        ds2.to_hdf5(filename=name2)
+        size1 = os.path.getsize(name1)
+        size2 = os.path.getsize(name2)
+        # If coords were stored per-entry then we would need a factor close to 2
+        # here. 1.1 implies that there is no such full duplication.
+        assert size1 * 1.1 > size2
+        # Empirically determined extra size. This is likely brittle. We want to
+        # somehow ensure that no column is stored more than once.
+        extra = 75000
```

Comment:
Can you add a bit more on where the extra size is from? I am guessing below the `len(coords) + 1` comes from the list of coords + 1 for the data, times the number of rows, time 8 for float64. So the extra is from creating a h5 file? Does this size change if we change the number of rows or coords?

Reply:
I don't really know, it is empirically determined. It does not depend on row count (added a comment).